torchao/quantization/quant_api.py

```diff
@@ -665,9 +666,8 @@ def apply_int4_weight_only_quant(weight):
         quant_min = 0
         quant_max = 15
         eps = 1e-6
-        preserve_zero = False
+        preserve_zero = zero_point_domain == ZeroPointDomain.INT
         zero_point_dtype = torch.bfloat16
-        zero_point_domain = ZeroPointDomain.FLOAT
 
         # Sparse Marlin only supports symmetric quantization.
         # NOTE: If we start having lots of layouts that require different configurations,
```

Comment:
might be better to add some assert for each of the layout for now, otherwise people can get invalid configurations, e.g. right now we have:
```
TensorCoreTiledLayout: zero point domain has to be FLOAT, and preserve_zero has to be False
MarlinSparseLayout: zero point has to be int, preserve zero has to be True
IntCPULayout: same as TensorCoreTiledLayout?
...
```